### PR TITLE
TSK-1909: optimize authorization in db2 query

### PR DESCRIPTION
**Pull request by ryzheboka**
Sonarcloud: https://sonarcloud.io/summary/new_code?id=ryzheboka_taskana&branch=TSK-1909
Both codesmells are unavoidable

---
 
### For the submitter:
- [ ] I updated the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) and will supply links to the specific files
- [x] I did not update the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview)
- [x] I included a link to the [SonarCloud branch analysis](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1019969636/SonarCloud+Integration)
- [ ] I added a description of changes on the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [x] I did not update the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [x] I put the ticket in review
- [ ] After integration of the pull request, I verified our [bluemix test environment](http://taskana.mybluemix.net/taskana) is not broken

### Verified by the reviewer:
- [ ] Commit message format → TSK-XXX: Your commit message.
- [ ] Submitter's update to [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) is sufficient
- [ ] SonarCloud analysis meets our standards
- [ ] Update of the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana) reflects changes
- [ ] PR fulfills the ticket
- [ ] Edge cases and unwanted side effects are tested
- [ ] Readability

### CODE REVIEW

File: lib/taskana-core-test/src/test/java/acceptance/task/query/TaskQueryImplAccTest.java [hunk 1]
```diff
@@ -149,6 +149,15 @@ void should_ReturnAllTasksAsTaskadmin_When_NoAccessItemForAWorkbasketExists() {
...
 
+    @WithAccessId(user = "admin")
+    @Test
+    void should_CountAllTasksFromWorkbasketAsAdmin_When_NoAccessItemForWorkbasketExists() {
+      long result =
+          taskService.createTaskQuery().workbasketIdIn(wbWithoutPermissions.getId()).count();
+
+      assertThat(result).isEqualTo(3);
+    }
+
```
Comment (mustaphazorgati) [inline lib/taskana-core-test/src/test/java/acceptance/task/query/TaskQueryImplAccTest.java:160]: why did you add this test? TSK-1909 is only a performance optimization. Shouldn't there be already a test which verifies this?

**Review by mustaphazorgati — CHANGES_REQUESTED** [review]

File: history/taskana-simplehistory-provider/src/test/java/acceptance/query/QueryWorkbasketHistoryAccTest.java [hunk 2]
```diff
@@ -67,6 +68,7 @@ void should_ReturnHistoryEvents_For_ComplexQuery() {
     assertThat(results).hasSize(3);
   }
 
+  @Disabled
```
[the remark this answers is not in the document]
Comment (ryzheboka) [reply history/taskana-simplehistory-provider/src/test/java/acceptance/query/QueryWorkbasketHistoryAccTest.java:71]: Ok :)

**Review by mustaphazorgati — APPROVED** [review]
